client/my-sites/customer-home/controller.jsx

```diff
@@ -34,21 +35,39 @@ export default async function ( context, next ) {
 export async function maybeRedirect( context, next ) {
 	const state = context.store.getState();
 	const slug = getSelectedSiteSlug( state );
-
 	if ( ! canCurrentUserUseCustomerHome( state ) ) {
 		page.redirect( `/stats/day/${ slug }` );
 		return;
 	}
 
 	const { verified, courseSlug } = getQueryArgs() || {};
 
+	const siteId = getSelectedSiteId( state );
+	const status = getSiteMigrationStatus( state, siteId );
+
+	if ( status === 'pending' ) {
+		page.redirect(
+			`/setup/site-migration/site-migration-how-to-migrate?siteSlug=${ slug }&siteId=${ siteId }`
+		);
+		return;
+	} else if ( status === 'pending-difm' ) {
+		page.redirect(
+			`/setup/site-migration/site-migration-credentials?siteSlug=${ slug }&siteId=${ siteId }`
+		);
+		return;
+	} else if ( status === 'pending-diy' || status === 'inactive' ) {
```

Comment:
I also checked `inactive` here because of the old DIY flow. If you get to the MigrateGuru screen, it will start a migration, and the API will return `inactive` as the status until the user starts the migration.